helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -1152,20 +1151,31 @@ private boolean updateControllerState(NotificationContext changeContext, PauseSi
 
     ClusterEventProcessor(BaseControllerDataProvider cache,
         ClusterEventBlockingQueue eventBlockingQueue, String processorName) {
-      super("HelixController-pipeline-" + processorName);
       _cache = cache;
       _eventBlockingQueue = eventBlockingQueue;
       _processorName = processorName;
     }
 
+    private String getThreadName(String eventId) {
+      String threadName = "HelixController-pipeline-" + _processorName;
+      if (!eventId.isEmpty()) {
+        threadName += "(" + eventId + ")";
```

Comment:
An extra String object would be created if going into this branch.  We can still use `StringBuilder` to reduce string garbage created because of temporary String as a good habit? Though this `getThreadName()` may not be called frequently and the performance difference is negligible.

Reply:
It's a tiny difference and got easy fixed.